### PR TITLE
Improve UX and HIG-conformity

### PR DESCRIPTION
**Pull request by real-yfprojects**
Make text in vorta HIG conform. I mainly followed the KDE HIG.
This PR not only changes punctuation and capitalization but also wording. Therefore reviewers should check whether they are ok with every change made.

This PR also tries to tackle all the tasks in #1050. They are listed below.

* [x]  Remove dots in misc tab
* [x]  Add colon to "Profile Name" in add profile window
* [x]  In schedule tab, replace "Prune old Archives after each backup" by  "Prune old archives after each backup"
* [x]  tool tip must follow the Header or sentence capitalization  rule .
* [x]  remove contraction: "Can't delete the last profile." -> "Cannot delete the last profile" (remove the dot !)... 
* [x]  Unlink repository button tooltip is too long.
* [x]  "Sentences should not be constructed from text in several controls. Sentences that run from one control to another will often not make sense when translated into other languages." (see wiki bellow for more explanation) :
  * [x]  Reformulate checkboxes text in Schedule>Clock
  * [x]  Reformulate checkboxes text in Archives>Prunes
* [x]  "clock", "wifi", "tasks" and "terminal" must be replace by "Clock", "Wifi", "Tasks'" and "Terminal"
* [x]  Replace three dots by ellipsis (see bellow)
* [x]  Put font 10 in profile add window. Try to shorten the text
* [x]  In source tab, add gradients bottoms '+/-' at bottom of the view.
* [x]  In Misc, items can be aggregate with labels : "Notifications", "Startup",  "Security", "Informations", "Others"
* [x]  Make layout more clear (related to the idea of advanced settings)
   * [x]  In Archive>Prunes
   * [x]  In Repository, SSH Key and Compression is sth. advanced

Additional tasks:
- [x] use affirmative phrases rather than use negative phrases for checkboxes
   - [x] `Scheduletab` > run over metered connections
- [x] `Scheduletab`>`shell commands`: Placeholder shouldn't be replacement for label
- [x] Source view: Improve UX of add and paste buttons
- [x]  #933 
- [x] Fix inconsistency in Schedule tab
- [x] Improve usability of archive view
- [x] Fix tooltips
- [x] context menus
    - [x] source view
    - [x] archive view
- [x] #1074

### CODE REVIEW

**Comment by real-yfprojects** [comment]
@bastiencyr Please have a look whether this meets your expectations regarding #1050.

**Comment by real-yfprojects** [comment]
Can you send some MacOS screenshots of this branch, @m3nu?

**Comment by m3nu** [comment]
That's a big and ambitious PR. Many good parts, but also some rough edges. Added a bunch of comments.

**Comment by real-yfprojects** [comment]
> Added a bunch of comments.

Where?

**Comment by m3nu** [comment]
> > Added a bunch of comments.
> 
> Where?

Here in the PR. Should be in this thread or in the "Files changed" tab.

**Comment by real-yfprojects** [comment]
> Should be in this thread or in the "Files changed" tab.

Something must have gone wrong. There aren't any.

**Comment by m3nu** [comment]
I see them while logged in, but not in an incognito window.

**Comment by m3nu** [comment]
They should be there now. Looks like I had to “submit” the review before the comments showed up.

**Comment by real-yfprojects** [comment]
> They should be there now. Looks like I had to “submit” the review before the comments showed up.

Ah, yes. Now they are there.

File: src/vorta/assets/UI/archivetab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/archivetab.ui:63]: There are two 'refreshes' one with the whole repo passed as an argument and one with only a single archive as argument.

File: src/vorta/assets/UI/archivetab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/archivetab.ui:54]: I thought, I fixed that but I will look into it again.

File: src/vorta/assets/UI/exportwindow.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/exportwindow.ui:23]: This is possible. I didn't want to repeat the checkbox label.

File: src/vorta/assets/UI/scheduletab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/scheduletab.ui:712]: This is puzzling because I set their horizontal size policy to `Expanding` and that is what they do on my machine.

File: src/vorta/assets/UI/repotab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/repotab.ui:71]: This is also set to expanding and works on Gnome and KDE.

File: src/vorta/assets/UI/scheduletab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/scheduletab.ui:727]: Can you check whether this is still the case? If not, can play around with qt designer a little bit to fix this?

This is also set to expanding and works on Gnome and KDE.

File: src/vorta/assets/UI/sourcetab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/sourcetab.ui:111]: It seems like Qt on MacOS doesn't support `QToolbuttons` but only `QPushbuttons`.

File: src/vorta/assets/UI/scheduletab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/scheduletab.ui:80]: I increased the main windows height.

File: src/vorta/assets/UI/archivetab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/archivetab.ui:54]: Should be fixed now.

File: src/vorta/assets/UI/scheduletab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/scheduletab.ui:749]: Can you check whether this is still the case? If not, can play around with qt designer a little bit to fix this?

File: src/vorta/assets/UI/sourcetab.ui
Comment (real-yfprojects) [reply src/vorta/assets/UI/sourcetab.ui:111]: Or it might relate to this note in the Qt docs:

> On macOS when a push button's width becomes smaller than 50 or its height becomes smaller than 30, the button's corners are changed from round to square. Use the setMinimumSize() function to prevent this behavior.

**Comment by m3nu** [comment]
Can you rebase so we can continue here? Thanks!

**Comment by real-yfprojects** [comment]
I fixed the merge conflict (and merged). There are still some comments of mine that you didn't answer yet.

**Comment by m3nu** [comment]
Thanks. Will continue with the comments shortly.

**Comment by m3nu** [comment]
I'll comment based on screenshots, since it's not easy to find the right place in those XMLs:

Screen Shot 2022-03-12 at 12 40 09

- Repository should use the full width.
- Mix of QToolbox and QButton. Would keep QToolbox, since that's what we use mostly.
- This screen looks pretty empty. Extra space could be used to guide new users (1. create ssh key, add repo, ...)
- Mix of different font sizes

**Comment by m3nu** [comment]
Screen Shot 2022-03-12 at 12 40 34

- Vertical scrolling in first and last section
- Why not use horizontal space?

**Comment by m3nu** [comment]
Screen Shot 2022-03-12 at 12 40 43

- Mix of QToolbox and buttons again.

Screen Shot 2022-03-12 at 12 49 45

- This screen seems very chaotic and all text boxes could be much wider, given their content.

**Comment by m3nu** [comment]
Added some comments. A few things left, but moving in the right direction!

**Comment by real-yfprojects** [comment]
> Repository should use the full width, since the URL (or maybe in the future name + URL) could be longer.
> This screen seems very chaotic and all text boxes could be much wider, given their content.

I can't fix that since it only occurs on MacOS which I don't have access to.
You can use 
```bash 
> git remote add yfprojects https://github.com/real-yfprojects
> git fetch yfprojects
> git branch hig-conform-text yfprojects/hig-conform-text
> git push --set-upstream m3nu hig-conform-text
```
to do that or the gihub web ui.

**Comment by real-yfprojects** [comment]
> Mix of QToolbox and QButton. Would keep QToolbox, since that's what we use mostly.

I'd use QButton because it uses the MacOS native style while QToolBox doesn't.

**Comment by real-yfprojects** [comment]
> No need to change the text below repository. It should communicate that there are local and remote repos and that BorgBase is one option for remote repos.
> This screen looks pretty empty. Extra space could be used to guide new users a bit more

I tried to shorten the text because the [KDE HIG](https://develop.kde.org/hig/style/writing/labels/) states:
> Keep labels short; be aware that [translated](https://develop.kde.org/hig/style/writing/localization) English text can expand up to 50% in some languages.
>
> Avoid static text: long instructions or explanations within the user interface. Being tempted to add static text is a good sign that the software’s user interface is too complex and should be simplified.

**Comment by real-yfprojects** [comment]
> We currently have some actions inside dropdowns. For SSH and Repos. Could make sense to make this more consistent and use it to QToolbox buttons?

That is a good idea.

**Comment by real-yfprojects** [comment]
> Mix of different font sizes

I have difficulties spotting them. Which labels are you referring to?

**Comment by m3nu** [comment]
> I have difficulties spotting them. Which labels are you referring to?

Below repo and below compression seems different here.
Screen Shot 2022-03-12 at 13 38 48

**Comment by m3nu** [comment]
> I'd use QButton because it uses the MacOS native style while QToolButton doesn't.

Please stick with QToolButton elements, since we already use them in most places and they need less space than buttons.

**Comment by m3nu** [comment]
No need. Will be squashed when merging.

**Comment by real-yfprojects** [comment]
> Why not use horizontal space?

grafik

or

grafik

**Comment by real-yfprojects** [comment]
Now these issues are left:

- Vertical scrolling on some screens
- Entries that do not expand on MacOS

You also didn't answer on the topic of static text in vorta.

**Comment by m3nu** [comment]
> Now these issues are left:
> 
> * Vertical scrolling on some screens
> * Entries that do not expand on MacOS
> 
> You also didn't answer on the topic of static text in vorta.

Thanks for the changes! Looks good to me, really. Thanks for the hard work! As you know I'm generally skeptical of very large PRs, but the latest changes look pretty good.

Can you adjust to fix the vertical scrolling, as you show in the screenshot? (both options are OK, just pick one you like)

I'll look into the macOS expansion before merging. (if you have "allow edits by maintainers" enabled for this PR)

**Comment by real-yfprojects** [comment]
> Can you adjust to fix the vertical scrolling, as you show in the screenshot? (both options are OK, just pick one you like)

I have chosen the second option because adding a manual line break into a checkbox label appears dirty to me. I already pushed the commit but github is struggling at the moment and doesn't update this thread.

> I'll look into the macOS expansion before merging. (if you have "allow edits by maintainers" enabled for this PR)

Enabled it.

**Comment by m3nu** [comment]
> I have chosen the second option because adding a manual line break into a checkbox label appears dirty to me.

The pristine solution gives me horizontal scrolling here. 😬

Will play with it today and then push it back without merging yet for you to check again.

**Comment by m3nu** [comment]
Added some changes for macOS. Hope this doesn't ruin it or mess it up on Linux:

Repsitory Tab:
- add min width to repo dropdown

Schedule tab:
- Make wording slightly shorter to avoid horizontal scrolling
- Use different element to expand line for shell commands

Screenshots:
Screen Shot 2022-03-23 at 12 05 43
Screen Shot 2022-03-23 at 12 05 53
Screen Shot 2022-03-23 at 12 05 19

**Comment by real-yfprojects** [comment]
Looks the same on Gnome and KDE. I will change the `sizeAdjustPolicy` of the comboBoxes to `AdjustToContents` in _repotab.ui_ and _mainwindow.ui_.

**Comment by m3nu** [comment]
Edited a few more minor issues:

- Some new SVG icons had a color specified, so override didn't work right away.
- A few button translations had a second string. Maybe should have been a text domain, which we don't use like this IIRC.
- Remove hardcoded size from profile dialog
- BorgBase link was missing openExternal attribute

From my side we can merge it with those changes.

**Comment by real-yfprojects** [comment]
> From my side we can merge it with those changes.

I agree.

**Review by m3nu — APPROVED** [review]

**Comment by m3nu** [comment]
The PR is big enough to get it's own tweet: https://twitter.com/borg_base/status/1507363578202005506

**Comment by real-yfprojects** [comment]
Have you updated transifex already? I think pushing a new source to transifex could be automated whenever a commit is added to master.

**Comment by m3nu** [comment]
Yes, pushed it yesterday, but not many new strings. As expected. Just existing string in new places.

And yes, could be added to CI. But not a big burden to run make while merging.

**Comment by real-yfprojects** [comment]
Do use the web UI of transifex or Qt Linguist for translations? Qt Linguist always displays the context of a string which I find very helpful.

**Comment by m3nu** [comment]
I use the commands in our `Makefile`:

```
make translations-from-source
make translations-push
```

More here: https://vorta.borgbase.com/contributing/translations/#using-and-testing-transifex-translations

**Comment by m3nu** [comment]
Never translated anything locally, so no idea. Currently translators add their translations on Transifex’ website, not via a local .ts files.

**Comment by real-yfprojects** [comment]
I think there is only poor support for translating locally when using transifex. This makes translating a tedious process if one wants to get the translation context right because one always has to open the corresponding file and the string inside it. It seems like there is nothing we can do about that.

**Pull request merged**
